contracts/carbon/Strategies.sol

```diff
@@ -514,17 +522,27 @@ abstract contract Strategies is Initializable {
     /**
      * @dev calculates the required amount plus fee
      */
-    function _addFee(uint128 amount) private view returns (uint128) {
+    function _addFee(uint128 amount, uint128 pairId) private view returns (uint128) {
+        uint32 tradingFeePPM = _getTradingFeePPM(pairId);
         // divide the input amount by `1 - fee`
-        return MathEx.mulDivC(amount, PPM_RESOLUTION, PPM_RESOLUTION - _tradingFeePPM).toUint128();
+        return MathEx.mulDivC(amount, PPM_RESOLUTION, PPM_RESOLUTION - tradingFeePPM).toUint128();
     }
 
     /**
      * @dev calculates the expected amount minus fee
      */
-    function _subtractFee(uint128 amount) private view returns (uint128) {
+    function _subtractFee(uint128 amount, uint128 pairId) private view returns (uint128) {
+        uint32 tradingFeePPM = _getTradingFeePPM(pairId);
         // multiply the input amount by `1 - fee`
-        return MathEx.mulDivF(amount, PPM_RESOLUTION - _tradingFeePPM, PPM_RESOLUTION).toUint128();
+        return MathEx.mulDivF(amount, PPM_RESOLUTION - tradingFeePPM, PPM_RESOLUTION).toUint128();
+    }
+
+    /**
+     * @dev get the custom trading fee ppm for a given pair (returns default trading fee if not set for pair)
+     */
+    function _getTradingFeePPM(uint128 pairId) private view returns (uint32) {
```

Comment:
Minor: I think that we're trying to avoid "get" as part of function naming (although I don't have a better suggestion here).